internal/services/network/network_interface_resource.go

```diff
@@ -386,14 +398,27 @@ func resourceNetworkInterfaceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		payload.Properties.IPConfigurations = ipConfigs
 	}
 
-	if d.HasChange("tags") {
+	if d.HasChange("tags") && !attachedToPrivateEndpoint {
 		tagsRaw := d.Get("tags").(map[string]interface{})
 		payload.Tags = tags.Expand(tagsRaw)
 	}
 
-	err = client.CreateOrUpdateThenPoll(ctx, *id, *payload)
-	if err != nil {
-		return fmt.Errorf("updating %s: %+v", *id, err)
+	if propsOtherThanTagsUpdated || !attachedToPrivateEndpoint {
```

Comment:
```suggestion
	if propsOtherThanTagsUpdated && !attachedToPrivateEndpoint {
```
I assume here should be `&&`? Otherwise, with regard to the scenario where a NIC does not have PE associated and it has no properties getting changed in the TF config, then the expectation is no API would be called while here a PUT{} would still be called which is unnecessary.

Reply:
@mybayern1974 I cannot use `&&` here, the tags will not be updated for the following scenario:

```
d.HasChange("tags") == true
propsOtherThanTagsUpdated == false
!attachedToPrivateEndpoint == true
```